tests/test_models/test_pendulum.py

```diff
@@ -0,0 +1,74 @@
+import math
+from typing import List
+
+import numpy as np
+import pytest
+from numpy.testing import assert_array_almost_equal, assert_array_almost_equal_nulp
+
+from hamilflow.models.pendulum import Pendulum, PendulumIC, PendulumSystem
+
+_LIS_OMEGA0 = [0.3, 0.6, 1.5]
```

Comment:
Would it be better to use pytest fixtures?

If we define a fixture for theta0, the lines

https://github.com/kausalflow/hamilflow/blob/a7426b090b94019a6fd50e176084be21e4e3e6f9/tests/test_models/test_pendulum.py#L34-L37

can be simplified to

```python
    def test_k(self, theta0: float) -> None:
        ic = PendulumIC(theta0=theta0)
        assert ic.k == np.sin(ic.theta0 / 2)
```

Reply:
I am not familiar with fixtures. How can I use them? I did not find fixtures in your code for testing.

Reply:
Not sure which is more elegant. I would use `TestPendulumSystem` as an example.

This is the current test.

```python
class TestPendulumSystem:
    @pytest.mark.parametrize("omega0", [-1.0, 0])
    def test_omega0_range(self, omega0: float) -> None:
        with pytest.raises(ValueError):
            _ = PendulumSystem(omega0=omega0)

    def test_init(self) -> None:
        _ = PendulumSystem(1)
```

With fixtures, it would become something like

```pytyhon
@pytest.fixture(
    params=[0.3, 0.6, 1.5]
)
def omega0(request):
    return request.param


class TestPendulumSystem:
    def test_omega0_range(self, omega0: float) -> None:
        with pytest.raises(ValueError):
            _ = PendulumSystem(omega0=omega0)

    def test_init(self) -> None:
        _ = PendulumSystem(1)
```

This is beneficial if we reuse the parameter `omega0` a lot as we don't need to repeat ourselves and write the line `@pytest.mark.parametrize(...)`.

Reply:
https://github.com/kausalflow/hamilflow/pull/22/commits/17c2db272d6159f026cbbcab73d1dfafac5c508c